sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -224,20 +224,53 @@
    * along with {@link Coder}s for (optional) key and values.
    */
   public static Write<byte[], byte[]> write() {
-    return new Write<byte[], byte[]>(
-        null,
-        ByteArrayCoder.of(),
-        ByteArrayCoder.of(),
-        TypedWrite.DEFAULT_PRODUCER_PROPERTIES);
+    return new AutoValue_KafkaIO_Write.Builder<byte[], byte[]>()
+        .setKeyCoder(ByteArrayCoder.of())
+        .setValueCoder(ByteArrayCoder.of())
+        .setProducerConfig(Write.DEFAULT_PRODUCER_PROPERTIES)
+        .build();
   }
 
   ///////////////////////// Read Support \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\
 
   /**
-   * A {@link PTransform} to read from Kafka topics. See {@link KafkaIO} for more
-   * information on usage and configuration.
+   * A {@link PTransform} to read from Kafka topics. See {@link KafkaIO} for more information on
+   * usage and configuration.
    */
-  public static class Read<K, V> extends TypedRead<K, V> {
+  @AutoValue
+  public abstract static class Read<K, V>
+      extends PTransform<PBegin, PCollection<KafkaRecord<K, V>>> {
+    abstract Map<String, Object> getConsumerConfig();
```

Comment:
They weren't before this PR; don't see a need to expand the API surface as part of this refactoring.